File: scripts/test_gemini.js

const https = require('https');

const API_KEY = 'AIzaSyC9ZUZhe_U6z5fKqplhYW7upzc_0oFapJY'; // New User Key
const CANDIDATES = [
    'gemini-1.5-flash',
    'gemini-1.5-pro',
    'gemini-pro',
    'gemini-2.0-flash-exp',
    'gemini-2.0-flash'
];

function testModel(model) {
    return new Promise((resolve) => {
        const url = `https://generativelanguage.googleapis.com/v1beta/models/${model}:generateContent?key=${API_KEY}`;
        const data = JSON.stringify({
            contents: [{ parts: [{ text: "Hello" }] }]
        });

        const req = https.request(url, {
            method: 'POST',
            headers: {
                'Content-Type': 'application/json',
                'Content-Length': data.length
            }
        }, (res) => {
            let body = '';
            res.on('data', c => body += c);
            res.on('end', () => {
                if (res.statusCode === 200) {
                    console.log(`[SUCCESS] ${model} worked!`);
                    resolve(model);
                } else {
                    console.log(`[FAILED] ${model} returned ${res.statusCode}`);
                    resolve(null);
                }
            });
        });

        req.on('error', (e) => {
            console.log(`[ERROR] ${model}: ${e.message}`);
            resolve(null);
        });

        req.write(data);
        req.end();
    });
}

async function run() {
    console.log("Testing generation with NEW KEY...");
    for (const m of CANDIDATES) {
        const result = await testModel(m);
        if (result) return;
    }
    console.log("All attempts failed.");
}

run();

File: test_gemini.py
import requests
import json

API_KEY = 'AIzaSyB7LoHWyfevhQalfe-nlrZgm_xPoSwL_KY'
URL = f'https://generativelanguage.googleapis.com/v1beta/models/gemini-2.5-flash:generateContent?key={API_KEY}'

headers = {
    'Content-Type': 'application/json'
}

data = {
    "contents": [{
        "parts": [{
            "text": "Hello, explain how AI works in one sentence."
        }]
    }]
}

print(f"Testing Gemini API with key: {API_KEY[:5]}...{API_KEY[-5:]}")
print(f"URL: {URL}")

try:
    response = requests.post(URL, headers=headers, json=data)
    
    print(f"Status Code: {response.status_code}")
    print("Response Body:")
    try:
        print(json.dumps(response.json(), indent=2))
    except:
        print(response.text)

except Exception as e:
    print(f"An error occurred: {e}")
